### DISCUSSION TITLE
golang projects are slow to compile

### DISCUSSION THREAD

**Discussion by wangxingzhen** [apache/skywalking#11779] [Q&A] [answered]
Do I need to carry -a every time I compile code in golang?  I found compilation to be too slow.  Before, it took about 10 seconds to compile the project, and after adding the -a command, it took a minute and 20 seconds. My server configuration is also a bit low, 2c 4G.But the speed difference is too much

**Comment by wu-sheng** [comment, marked as answer]
Yes, you have to. The tracing and instrumentation codes are only weaved into your application codes through compiling. There is no magic.

**Reply by wangxingzhen** [reply]
OK
